tests/vasp/test_handlers.py

```diff
@@ -1016,6 +1016,24 @@ def test_check_and_correct(self) -> None:
             assert content == "LABORT = .TRUE."
         os.remove("STOPCAR")
 
+    def test_check_with_malformed_outcar(self, tmp_path: Path) -> None:
+        """Test that WalltimeHandler.check() returns False on malformed OUTCAR.
+
+        This can happen when VASP is mid-write and the file contains incomplete
+        data like a standalone '-' instead of a float. See
+        https://github.com/materialsproject/pymatgen/issues/2251
+        """
+        os.chdir(tmp_path)
```

Comment:
@janosh: FYI that better practice here would be to do `monkeypatch.chdir(tmp_path)` to not modify the path of the main process in the test suite.

Reply:
Never mind. I see this is addressed by the `tearDown` call and is done throughout the module.

Reply:
strongly agree. been meaning to overhaul the `custodian` test suite for a long time specifically to fix this but no time